### ISSUE TITLE
Inconsistent errors with nested verbatim interpolated string literals (@$ vs $@ CS8087)

### ISSUE THREAD

**Issue by JeffreyMercado** [dotnet/roslyn#44587]
Verbatim interpolated string literals using `@$` or `$@` nested within an interpolated string are not handled consistently. They should be equivalent but one has an error (CS8087) if a literal backslash `\` precedes a `{` in the `@$` string. It doesn't matter what form the outer interpolated string is.

```c#
var value = "foo";

//var a0 = $"\{value}"; // error, CS8087 (expected)
var a1 = $"\\{value}"; // ok
var a2 = $@"\{value}"; // ok
var a3 = @$"\{value}"; // ok

//var b0 = $"\\{$"\{value}"}"; // error, CS8087 (expected)
var b1 = $"\\{$"\\{value}"}"; // ok
var b2 = $"\\{$@"\{value}"}"; // ok
var b3 = $"\\{@$"\{value}"}"; // error, CS8087

//var c0 = $@"\{$"\{value}"}"; // error, CS8087 (expected)
var c1 = $@"\{$"\\{value}"}"; // ok
var c2 = $@"\{$@"\{value}"}"; // ok
var c3 = $@"\{@$"\{value}"}"; // error, CS8087

//var d0 = @$"\{$"\{value}"}"; // error, CS8087 (expected)
var d1 = @$"\{$"\\{value}"}"; // ok
var d2 = @$"\{$@"\{value}"}"; // ok
var d3 = @$"\{@$"\{value}"}"; // error, CS8087
```

**Version Used**: `.Net Core 3.1.201` and `Roslyn 3.5.0-beta4-20153-05`

**Steps to Reproduce**: See example

**Expected Behavior**: No errors in the example

**Actual Behavior**: CS8087 for b3, c3, and d3.

> CS8087 A '{' character may only be escaped by doubling '{{' in an interpolated string.

**Comment by CyrusNajmabadi** [comment]
> Actual Behavior: CS8087 for b3, c3, and d3.

Not reproing this:

<img width="1182" height="918" alt="Image" src="https://github.com/user-attachments/assets/71070642-0b46-4a50-a6ec-f08b4e7761bb" />

This all seems correct.
